spark/src/test/scala/org/apache/spark/sql/delta/stats/DataSkippingDeltaTests.scala

```diff
@@ -599,6 +599,31 @@ trait DataSkippingDeltaTestsBase extends DeltaExcludedBySparkVersionTestMixinShi
     deltaStatsColNamesOpt = Some("b.c")
   )
 
+  testSkipping(
+    "indexed column names - naming a nested column allows nested complex types",
+    """{
+      "a": {
+        "b": [1, 2, 3],
+        "c": [4, 5, 6],
+        "d": 7,
+        "e": 8,
+        "f": {
+          "g": 9
+        }
+      },
+      "i": 10
+    }""".replace("\n", ""),
+    hits = Seq(
```

Comment:
Could we have hit tests for elements inside struct and the double struct as well?

Btw click on re-request review so that I get notified.

Reply:
Added more hit tests

Reply:
My understanding is that when we are inside struct, we gather stats for the valid datatypes inside the struct, but the invalid ones we still don't right? Would we want to test that we still don't skip the invalid datatypes inside the struct?

Reply:
Also, if we can check another skipping valid type beside integer, like timestamp, string, ... that would be great

Reply:
Technically null counts are collected, but they can't be used for skipping. We could add that to the test, but seems kind of out of scope because you can't do min/max things with arrays/maps anyway, so it'd have to be like an element contains or something

Reply:
I think there are other non-skipping eligible datatypes than lists datatypes

Reply:
I'm not sure how to specify a Binary type in the inferred JSON. Is that the only non-complex type?